ndcube/visualization/descriptor.py

```diff
@@ -18,35 +20,76 @@ def __set_name__(self, owner, name):
         self._attribute_name = f"_{name}"
+        plotter = self._resolve_type()
+        if hasattr(plotter, "plot"):
+            functools.update_wrapper(owner.plot, plotter.plot)
 
-    def __get__(self, obj, objtype=None):
-        if obj is None:
-            return
 
-        if getattr(obj, self._attribute_name, None) is None:
-
-            # We special case the default MatplotlibPlotter so that we can
-            # delay the import of matplotlib until the plotter is first
-            # accessed.
+    def _resolve_default_type(self, obj):
+        # We special case the default MatplotlibPlotter so that we can
+        # delay the import of matplotlib until the plotter is first
+        # accessed.
+        if self._default_type in ("mpl_plotter", "mpl_sequence_plotter"):
             try:
                 if self._default_type == "mpl_plotter":
                     from ndcube.visualization.mpl_plotter import MatplotlibPlotter
                     self.__set__(obj, MatplotlibPlotter)
                 elif self._default_type == "mpl_sequence_plotter":
                     from ndcube.visualization.mpl_sequence_plotter import MatplotlibSequencePlotter
                     self.__set__(obj, MatplotlibSequencePlotter)
-                elif self._default_type is not None:
-                    self.__set__(obj, self._default_type)
-                else:
-                    # If we have no default type then just return None
-                    return
             except ImportError as e:
                 raise ImportError(MISSING_MATPLOTLIB_ERROR_MSG) from e
 
+        elif self._default_type is not None:
+            self.__set__(obj, self._default_type)
+
+        # If we have no default type then just return None
+        else:
+            return
+
+    def _resolve_type(self):
+        # We special case the default MatplotlibPlotter so that we can
+        # delay the import of matplotlib until the plotter is first
+        # accessed.
+        if self._default_type in ("mpl_plotter", "mpl_sequence_plotter"):
+            try:
+                if self._default_type == "mpl_plotter":
+                    from ndcube.visualization.mpl_plotter import MatplotlibPlotter
+                    return MatplotlibPlotter
+                elif self._default_type == "mpl_sequence_plotter":
+                    from ndcube.visualization.mpl_sequence_plotter import MatplotlibSequencePlotter
+                    return MatplotlibSequencePlotter
+            except ImportError as e:
+                raise ImportError(MISSING_MATPLOTLIB_ERROR_MSG) from e
+
+        elif self._default_type is not None:
+            return self._default_type
+
+        # If we have no default type then just return None
+        else:
+            return
+
+    def __get__(self, obj, objtype=None):
+        if obj is None:
+            return
+
+        if getattr(obj, self._attribute_name, None) is None:
+            self._resolve_default_type(obj)
```

Comment:
We don't need both `_resolve_type` and `_resolve_default_type` we should make this use the current `_resolve_type` implementation (although I think `_resolve_default_type` is a better name).